app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt

```diff
@@ -21,6 +37,51 @@ class WelcomeFragment : Fragment() {
         rootView.pickCityButton.setOnClickListener {
             Navigation.findNavController(rootView).navigate(R.id.searchLocationFragment, null, Utils.getAnimSlide())
         }
+
+        geoLocationViewModel.currentLocationVisibility.observe(this, Observer {
+            rootView.currentLocation.visibility = View.GONE
+        })
+        geoLocationViewModel.configure(null)
+
+        rootView.currentLocation.setOnClickListener {
+            checkLocationPermission()
+            geoLocationViewModel.configure(activity)
+            rootView.locationProgressBar.visibility = View.VISIBLE
+        }
+        geoLocationViewModel.location.observe(this, Observer { location ->
+            searchLocationViewModel.saveSearch(location)
+            redirectToMain()
+        })
+
         return rootView
     }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        when (requestCode) {
+            LOCATION_PERMISSION_REQUEST -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    geoLocationViewModel.configure(activity)
+                } else {
+                    Snackbar.make(rootView, "Cannot fetch location!", Snackbar.LENGTH_SHORT).show()
+                    rootView.locationProgressBar.visibility = View.GONE
+                }
+            }
+        }
+    }
+
+    private fun checkLocationPermission() {
+        val permission =
+            ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION)
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST)
+        }
+    }
+
+    private fun redirectToMain() {
+        val startMainActivity = Intent(activity, MainActivity::class.java)
+        activity?.startActivity(startMainActivity)
```

Comment:
Same. Why don't you use navigation controller?

Reply:
Okay will do that.

Reply:
@liveHarshit On doing that I am losing the action bar as not provided in the welcome fragment. Any suggestions how to solve that?

Reply:
Sorry, cannot get it?

Reply:
Add `thisActivity.supportActionBar?.show()` inside `if (thisActivity is AppCompatActivity)` in onCreateView of events fragment.